### DISCUSSION TITLE
Influence of batch_size on running validation.

### DISCUSSION THREAD

**Discussion by motykatomasz** [Lightning-AI/pytorch-lightning#13090] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Recently I've observed different, weird behaviors during training vision models using PL (version 1.5.9):

- Callback "on_validation_epoch_end" was being called before the validation even happened.
- Validation was happening every ~15/20 epochs (no strict pattern) event though it was stated in the config that it should be called every epoch.

I was able to alleviate such issues by changing the batch_size slightly. I was wondering in what way the batch_size might influence such behavior. Has anyone else experienced similar issues?

**Comment by akihironitta** [comment]
@motykatomasz Still seeing the same issue in the stable release `1.6.4`? If so, could you provide a minimal script with [BoringModel](https://github.com/PyTorchLightning/pytorch-lightning/tree/master/pl_examples/bug_report) and your env details for a repro?

FYI, the hook order: https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#hooks
